ampd/src/starknet/json_rpc.rs

```diff
@@ -106,6 +114,58 @@ where
 
         Ok(event)
     }
+
+    /// Fetches a transaction receipt by hash and extracts a SignersRotatedEvent if present
+    ///
+    /// # Arguments
+    ///
+    /// * `tx_hash` - The hash of the transaction to fetch
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(Some((tx_hash, SignersRotatedEvent)))` - If the transaction exists and contains a valid SignersRotatedEvent
+    /// * `Ok(None)` - If the transaction exists but contains no SignersRotatedEvent
+    /// * `Err(StarknetClientError)` - If there was an error fetching the receipt or the transaction failed
+    ///
+    /// # Errors
+    ///
+    /// Returns a `StarknetClientError` if:
+    /// * Failed to fetch the transaction receipt from the node
+    /// * The transaction execution was not successful
+    async fn get_event_by_hash_signers_rotated(
```

Comment:
You can create tests for that by mocking the responses. Check how the other tests are made with for example the `InvalidContractCallEventMockTransport` and `ValidMockTransport`.

You can mock with [this starknet gateway SignersRotated event](https://sepolia.voyager.online/event/351561_3_0).

Reply:
Done